File: R/testChi.R
`testChi` <- function(nchi = 100000){


  dims = rep(0,1)
  dims[1] = nchi
  
 
  Chioutput = rep(0,nchi)

  
  Z = .C("testChiC", Chioutput=as.double(Chioutput),as.integer(dims), PACKAGE="SubsetRV")

  Chioutput= Z$Chioutput
  return(Chioutput)
}

File: src/testChi.c

#include <R_ext/Lapack.h>
#include <R_ext/Applic.h>
#include <R_ext/PrtUtil.h>
#include <stdio.h>
#include <stdlib.h>
#include <math.h>
#include <time.h>
#include <R.h>
#include <Rmath.h>
#include "utility.h"



/*********************************************************************
 *
 * 
 *
 *********************************************************************/
 
void testChiC(double* Chioutput,int* dims)
{
    int nchi = dims[0], j;
    GetRNGstate();
    for(j=0; j<nchi; j++){
      Chioutput[j] = rchisq(1);
    }
    PutRNGstate();
} 

